src/Camunda.Worker/CamundaWorkerBuilderExtensions.cs

```diff
@@ -36,7 +36,7 @@ private static HandlerMetadata CollectMetadataFromAttributes(Type handlerType)
             return new HandlerMetadata(topicsAttribute.TopicNames, topicsAttribute.LockDuration)
             {
                 LocalVariables = variablesAttribute?.LocalVariables ?? false,
-                Variables = variablesAttribute?.Variables,
+                Variables =  variablesAttribute?.AllVariables ?? false ? null : variablesAttribute?.Variables,
```

Comment:
Could you add the test for this behavior

Reply:
Sure